src/details/ArborX_DetailsMutualReachabilityDistance.hpp

```diff
@@ -47,34 +47,6 @@ struct MaxDistance
   }
 };
 
-template <class Primitives>
-struct NearestK
-{
-  Primitives primitives;
-  int k; // including self-collisions
```

Comment:
Do we have a comment at call site or anywhere saying the `k` argument includes self-collisions?

Reply:
I'm not sure why this comment was there in the first place. Nearest search always include self-points if the primitives and predicates are the same. I think this is more about the terminology in DBSCAN/HDBSCAN where `minPts` includes points themselves.